File: Coding/Ruby/guessing_game.rb
puts "Hello user! This is a game to guess the number between 1 and 100.\n"
number = rand(1..100)
repeat = 'y'
prev_guess = '/0'
tries = 0

while repeat == 'y'
    puts "Enter your guess now: "
    guess = gets.to_i
    
    if guess == number
        tries+=1
        puts "Congratulations! That is the correct answer. You guessed #{tries} times. Would you like to try again?"
        repeat = gets.chomp
        number = rand(1..100)
        prev_guess = '/0'
        tries = 0
    elsif guess == prev_guess
        puts "You guessed the same number again #{prev_guess}. Try again!\n"
    elsif guess < number
        tries+=1
        puts "Too small! #{tries} guesses so far.\n"
        prev_guess = guess
    elsif guess > number
        tries+=1
        puts "Too large! #{tries} guesses so far.\n"
        prev_guess = guess
    end
end

File: Coding/Perl/guessing_game.pl
# This is a guessing game to find the value of a random number between 1 and 100. The game will tell your current guess number and whether your guess was too high or low
# The user can also guess the same value two or more times without penalty

$number = int(rand(99)) + 1;
$repeat = "y";
$prev_guess = '/0';
$tries = 0;


while($repeat eq "y"){
    print "Please enter your guess now: ";
    chomp($guess=<>);
    $guess = int($guess);
    
    if( $guess== $number ){
        $tries++;
        print "Congratulation! You have guessed the correct number $number. It took you $tries to guess the correct answer.\n\n Would you like to play again?(y/any)";
        chomp($repeat=<>);
        $number = int(rand(99)) + 1;
        $prev_guess = '/0';
        $tries = 0;
    } elsif ( $guess == $prev_guess ) {
        print "You made the same guess in a row, $prev_guess. Try another number.\n";
    } elsif ( $guess < $number ) {
        $tries++;
        print " Too small! You've guessed $tries.\n";
        $prev_guess = $guess;
    } elsif ( $guess > $number ) {
        $tries++;
        print " Too large! You've guessed $tries.\n";
        $prev_guess = $guess;    
    }
}